### PR TITLE
Feature/testcase 169

### PR DESCRIPTION
**Pull request by jschleicher**
## Description
Add test case that fails before the fix of #169.
Since all tests are running in devel space, we have to create an own package which has `psen_scan_v2` as dependency and runs the launch file from the (isolated) install space (see https://answers.ros.org/question/241467/testing-launch-files-in-install_space/) Thus the main diff of this PR moves all files into a new subdirectory `psen_scan_v2` which is still our ROS package. Then a new test package is created, where CI fails:
```
[ROSUNIT] Outputting test results to /root/target_ws/build/test_psen_scan_v2_prerelease/test_results/test_psen_scan_v2_prerelease/rostest-test_integrationtest_ros_scanner_launch.xml
    /root/target_ws/install/psen_scan_v2/lib/psen_scan_v2/psen_scan_v2_node: error while loading shared libraries: libpsen_scan_v2_standalone.so: cannot open shared object file: No such file or directory
```
and finally a merge of the main branch succeeds with the fix.

Please squash-Merge keeping the commit message of e5413b6 and this description.

I'd suggest to ignore the new package `test_psen_scan_v2_prerelease` for the bloom release process as in https://answers.ros.org/question/66545/to-ignore-some-pkgs-upon-release-by-bloom/

### Things to add, update or check by the maintainers before this PR can be merged.

* [ ] ~Public api function documentation~
* [ ] ~Architecture documentation reflects actual code structure~
* [ ] ~[Tutorials](https://wiki.ros.org/pilz_robots/Tutorials/)~
* [ ] ~Overview on [ROS wiki](https://wiki.ros.org/pilz_robots)~
* [x] Package Readme ([example pilz_robots](https://github.com/PilzDE/pilz_robots/blob/melodic-devel/README.md))
* [x] Good commit messages ([some tips](https://dev.to/jacobherrington/how-to-write-useful-commit-messages-my-commit-message-template-20n9))
* [ ] ? CHANGELOG.rst updated
* [x] Copyright headers
* [ ] ~Examples~

### Review Checklist
* [ ] Soft- and hardware architecture (diagrams and description)
* [ ] Test review (test plan and individual test cases)
* [ ] Documentation describes purpose of file(s) and responsibilities
* [ ] Code (coding rules, style guide)

### Release planning (please answer)
* [x] When is the new feature released? With next regular release. No need to push this.
* [x] ~Which dependent packages do have to be released simultaneously?~

### CODE REVIEW

File: psen_scan_v2/CMakeLists.txt [hunk 1]
```diff
@@ -379,7 +379,7 @@ if(CATKIN_ENABLE_TESTING)
     ${catkin_LIBRARIES}
     ${pilz_testutils_LIBRARIES}
   )
-
+
```
Comment (martiniil) [inline psen_scan_v2/CMakeLists.txt:382]:
```suggestion

```

File: test_psen_scan_v2_prerelease/test/integrationtest_ros_scanner_launch.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import unittest
+import rospy
+
+class TestStartup(unittest.TestCase):
+    def test_startup(self):
+        i = 0
+        expectedTopic = ['/laser_scanner/scan', 'sensor_msgs/LaserScan']
+        while not expectedTopic in rospy.get_published_topics():
+            rospy.sleep(rospy.Duration(0.5))
+            self.assertLess(i, 10, msg="node did not come up in the expected time") # wait 5s max
```
Comment (martiniil) [inline test_psen_scan_v2_prerelease/test/integrationtest_ros_scanner_launch.py:12]: I would not rely on a topic being published without attached hardware. We could instead just check if the node comes up: https://github.com/PilzDE/pilz_robots/blob/8c570447a73730b2f5e8538104eb7b6418389472/prbt_hardware_support/test/include/prbt_hardware_support/ros_test_helper.h#L46

**Review by martiniil — CHANGES_REQUESTED** [review]
See inline comments.

**Comment by martiniil** [comment]
@jschleicher Thanks for your effort. We discussed about it and figured out that there is already a test that would have failed (https://github.com/PilzDE/psen_scan_v2/tree/main/test/hw_tests). We just need to move it into a separate package and get our test-area running.

I suggest to close here.

**Comment by jschleicher** [comment]
Maybe another launchfile-check without hardware would still be feasible? At least as long as the acceptance test isn't fully automated...?

**Comment by martiniil** [comment]
> 
> 
> Maybe this launchfile-check without hardware would still be feasible? At least as long as the acceptance test isn't fully automated...?

Without moving all the files: https://github.com/PilzDE/pilz_testutils/pull/26

**Pull request closed without merging**